mlops/bicep/modules/networking/vnet.bicep

```diff
@@ -0,0 +1,86 @@
+// Creates a virtual network with service endpoint for Storage
+
+targetScope = 'resourceGroup'
+
+@description('Azure region of the deployment')
+param location string = resourceGroup().location
+
+@description('Tags to add to the resources')
+param tags object = {}
+
+@description('Name of the virtual network resource')
+param virtualNetworkName string
+
+@description('Group ID of the network security group')
+param networkSecurityGroupId string
+
+@description('Virtual network address prefix')
+param vnetAddressPrefix string = '192.168.0.0/16'
+
+@description('Training subnet address prefix')
+param subnetPrefix string = '192.168.0.0/24'
+
+@description('Subnet name')
+param subnetName string = 'snet-training'
+
+
+// @description('Scoring subnet address prefix')
+// param scoringSubnetPrefix string = '192.168.1.0/24'
+
+resource virtualNetwork 'Microsoft.Network/virtualNetworks@2022-01-01' = {
+  name: virtualNetworkName
+  location: location
+  tags: tags
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        vnetAddressPrefix
+      ]
+    }
+    subnets: [
+      { 
+        name: subnetName
+        properties: {
+          addressPrefix: subnetPrefix
+          privateEndpointNetworkPolicies: 'Disabled'
+          privateLinkServiceNetworkPolicies: 'Disabled'
+          networkSecurityGroup: {
+            id: networkSecurityGroupId
+          }
+          // IMPORTANT: if you don't add this, you will not be able to add the storage
+          // to your subnet and will get an internal error
+          serviceEndpoints: [
+            {
+              service: 'Microsoft.Storage'
+            }
+          ]
+        }
+      }
+      // { 
```

Comment:
I suppose we're keeping this one in case we want to enable scoring too in the future?

Reply:
Just because I want to keep it to figure out later, I've added a comment inline to explain.